personality/se_dpm/puppet/puppetconf.pan

```diff
@@ -7,84 +7,92 @@ include 'components/puppet/config';
+variable GRIDFTP_REDIR_ENABLED ?= false;
+variable HTTPS_ENABLED ?= false;
+variable DPM_MEMCACHED_ENABLED ?= false;
+variable GRIDFTP_REDIR_ENABLED ?= false;
+variable DOME_ENABLED ?= false;
+variable DOME_FLAVOUR ?= false;
+variable DMLITE_TOKEN_PASSWORD ?= 'mytokenpassword';
+
+variable DPM_LOG_LEVEL ?= 0;
+variable DPM_DISK_LOG_LEVEL ?= DPM_LOG_LEVEL;
+variable DPM_HEAD_LOG_LEVEL ?= DPM_LOG_LEVEL;
 
 '/software/components/puppet/modules' ?= dict();
 
 '/software/components/puppet/modules' = {
-  if(DPM_PUPPET_MODULE != 'NONE'){
-    SELF[escape(DPM_PUPPET_MODULE)] = dict('version',DPM_PUPPET_MODULE_VERSION);
-  };
-  SELF;
+    if(DPM_PUPPET_MODULE != 'NONE'){
+        SELF[escape(DPM_PUPPET_MODULE)] = dict('version', DPM_PUPPET_MODULE_VERSION);
+    };
+    SELF;
 };
 
-variable DPMMGR_UID?=970;
-
-'/software/components/puppet/hieradata' = {
-	self=dict();
-	if(FULL_HOSTNAME==DPM_HOSTS['dpm'][0]){
-		self['classes'] = 'dpm::headnode';
-	}else{
-		self['classes'] = 'dpm::disknode';
-	};	
-	self[escape('dpm::params::localdomain')] = SITE_DOMAIN;
-	self[escape('dpm::params::headnode_fqdn')] = DPM_HOSTS['dpm'][0];
-
-	self[escape('dpm::params::volist')] = VOS;
-
-	if ( pkg_compare_version(DPM_PUPPET_MODULE_VERSION,'1.8.10') == PKG_VERSION_LESS ) {
-	  disk_list='';
-	  foreach(i;disk;DPM_HOSTS['disk']){
-	    if(disk_list==''){
-	      disk_list=disk;
-	    }else{
-	      disk_list=disk_list+' '+disk;
-	    };
-	  };
-        }else{
-          disk_list=DPM_HOSTS['disk'];
-	};
-
-	self[escape('dpm::params::disk_nodes')] = disk_list;
-
-	self[escape('dpm::params::dpmmgr_uid')] = DPMMGR_UID;
-	self[escape('dpm::params::dpmmgr_gid')] = 970;
-
-	self[escape('dpm::params::token_password')] = 'mytokenpassword';
-	self[escape('dpm::params::xrootd_sharedkey')] = DPM_XROOTD_SHARED_KEY;
-	self[escape('dpm::params::db_pass')] = DPM_DB_PARAMS['password'];
-
-	self[escape('dpm::params::mysql_root_pass')] = DPM_DB_PARAMS['adminpwd'];
-	self[escape('dpm::params::dpm_xrootd_fedredirs')] = XROOTD_FEDERATION_PARAMS;
-
-	if(is_defined(XROOTD_REPORTING_OPTIONS)&&is_defined(XROOTD_MONITORING_OPTIONS)){
-		self[escape('dpm::params::xrd_report')] = XROOTD_REPORTING_OPTIONS;
-		self[escape('dpm::params::xrootd_monitor')] = XROOTD_MONITORING_OPTIONS;
-	};
-
-	if(is_defined(XROOTD_SITE_NAME)){ 
-		self[escape('dpm::params::site_name')] = XROOTD_SITE_NAME;
-	};
-
-	if(is_defined(HTTPS_ENABLED) && HTTPS_ENABLED){ 
-		self[escape('dpm::params::webdav_enabled')] = true;
-		if(is_defined(DPM_MEMCACHED_ENABLED) && DPM_MEMCACHED_ENABLED){
-				self[escape('dpm::params::memcached_enabled')] = true;
-		};
-	};
-
-	if(is_defined(GRIDFTP_REDIR_ENABLED) && GRIDFTP_REDIR_ENABLED){
-		self[escape('dpm::params::gridftp_redirect')] = true;
-	};
-
-	if(is_defined(SPACE_REPORTING_ENABLED) && SPACE_REPORTING_ENABLED){
-		self[escape('dpm::params::enable_space_reporting')] = true;
-	};
-
-	self;
+variable DPMMGR_UID ?= 970;
+variable DPMMGR_GID ?= 970;
+
+function set_yaml_boolean = {
```

Comment:
Shouldn't this kind of conversion be handled by the component code?

Reply:
Well, the component just dumps json structure into yaml, it would be difficult to parse it, find out booleans and turn them into yes/no. Moreover, I'm not sure this apply everywhere. And also I'm not using it only for the booleans. Probably the function is misnamed, it is actualy a function returning a value or another depending on a boolean flag. Something similar to the 

exp ? value1 : value2 

construct (we do not have it in pan, do we?). In this sense it may be a generally useful function. Maybe it should be renamed properly and moved to some general function template.

Reply:
Ok fair enough.

BTW pan's `if` is designed be used in this way (remember that everything in pan returns a value).
```pan
'/foo/bar' = if (exp) value1 else value2;
```

Reply:
OK. Indeed I've not thought about using if in that way. I can re-write the profile using this one-line if construct instead of the set_yaml_boolean function, if your prefer.

Reply:
It's up to you, whichever you think is most readable.